### ISSUE TITLE
Strange Errors With Jade

### ISSUE THREAD

**Issue by joeyjoejoejr** [carrot/roots-contentful#27]
I'm getting the same strange errors mentioned here:
https://github.com/jenius/roots/pull/662

and that are seemingly fixed in roots records here:
https://github.com/carrot/roots-records/issues/22

I'll try to look into it when I get a chance, but I'm getting the same issue in roots-contentful

**Comment by kylemac** [comment]
@joeyjoejoejr is this happening on the initial compile or only on subsequent compiles? I've run into a problem with only subsequent compiles lately and I'm trying to figure out whether they are the same issue or not

**Comment by joeyjoejoejr** [comment]
Yeah, only subsequent compiles.  The fix should be in the repo.  We need to sanitize the locals that are passed into Jade.  I haven't had a chance to look at it yet.

**Comment by kylemac** [comment]
@joeyjoejoejr i'm not totally sure what you mean but based on what you say, it sounds like an issue here - https://github.com/carrot/roots-contentful/blob/master/lib/index.coffee#L137-L138

**Comment by joeyjoejoejr** [comment]
It's actually right here:
https://github.com/carrot/roots-contentful/blob/master/lib/index.coffee#L154-L155

Where the locals are being passed into renderFile.  The locals have some left-over data from roots that causes problems with the jade compiler.  Specifically, I think, is the filename properly.

**Comment by kylemac** [comment]
update for you @joeyjoejoejr - this one has been stickier than we expected. it has its hooks in accord and roots-core, but we're working on it. @hhsnopek has it in his sights

**Comment by declandewet** [comment]
Is this issue still persistent? /cc @kylemac @hhsnopek @joeyjoejoejr

**Comment by hhsnopek** [comment]
This was solved in roots - see the roots issue #662, for anyone that needs a working branch of roots that's still on v3.1.0 see: https://github.com/carrot/roots/tree/upstream-v3.1.0
